src/vs/workbench/browser/media/style.css

```diff
@@ -157,6 +157,10 @@ body.web {
 	padding-right: 3px;
 }
 
+.monaco-workbench:not(.file-icons-enabled) .product-icon[class*='codicon-']::before {
```

Comment:
I wonder if this change too aggressive? I was expecting something closer to where we hand the editor icon off to the icon label to control this?

Reply:
I considered that approach, but most of the ion theme stuff is handled with global CSS rules and I felt like this would guarantee that it would be fixed correctly everywhere.

Reply:
My initial worry was that the setting is about file icons being disabled or not but the CSS rule might impact other icons? I would run this with @aeschli to be sure if we want to do it this way.

Reply:
The `.product-icon` class name gives the impression that all product icons are disabled. Better use a more specific class name. `product-file-icon`, `predefined-file-icon`...